hydra/_internal/core_plugins/importlib_resources_config_source.py

```diff
@@ -2,12 +2,28 @@
 import os
 from typing import List, Optional
 
-import importlib_resources
 from omegaconf import OmegaConf
 
 from hydra.core.object_type import ObjectType
 from hydra.plugins.config_source import ConfigLoadError, ConfigResult, ConfigSource
 
+try:
+    import importlib_resources
+
+    # `importlib_resources` is avilable only till Python 3.8. Beyond this,
+    # one should use `importlib.resources`.
+    # Note that `importlib.resources` is available on Python 3.8 but its
+    # API does not match the API of `importlib_resources`. However, in
+    # Python 3.9, the API of `importlib.resources` matches the API of
+    # `importlib_resources`.
+    # So, by default, we try to use `importlib_resources` (supported till 3.8)
+    # and if that is not found, we use `importlib.resources`.
+    # Switching the order of import statements breaks the code for Python 3.8.
+except ModuleNotFoundError:
+    from importlib import resources as importlib_resources  # type: ignore
+
+    # This is a mypy bug: https://github.com/python/mypy/issues/1153
```

Comment:
```
if python version >= 3.9:
  from importlib import resources as importlib_resources 
else:
  import importlib_resources
```

Reply:
Resolved in  https://github.com/facebookresearch/hydra/pull/1197/commits/97aab6b6416923dc59ae6403cbc95de516480fb0 but retaining the comment (in case someone wastes their time trying to use `importlib.resources` for py 3.8